core-web/libs/block-editor/src/lib/shared/services/search/search.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { pluck } from 'rxjs/operators';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+export enum ESOrderDirection {
+    ASC = 'ASC',
+    DESC = 'DESC'
+}
+
+export interface queryEsParams {
+    itemsPerPage?: number;
+    filter?: string;
+    lang?: string;
+    offset?: string | number;
+    query: string;
+    sortField?: string;
+    limit?: number;
+    sortOrder?: ESOrderDirection;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SearchService {
+    constructor(private http: HttpClient) {}
+
+    /**
+     * Returns a list of contentlets from Elastic Search endpoint
+     * @param queryEsParams params
+     * @returns Observable<ESContent>
+     * @memberof DotESContentService
+     */
+    public get({ query, limit = 0, offset = 0 }: queryEsParams): Observable<unknown> {
```

Comment:
```suggestion
    public get<T>({ query, limit = 0, offset = 0 }: queryEsParams): Observable<T> {
```

We should use generics here.